src/02_spec/07_modules/cdm/index.rst

```diff
@@ -1,40 +1,28 @@
-Core Debug Module (CDM)
-=======================
+Core Debug Module for OpenRISC CPU Cores (CDM-OR1K)
+===================================================
 
-.. figure:: ../../img/debug_module_cdm.*
-   :alt: Core Debug Module
+.. figure:: ../../img/debug_module_cdm_or1k.*
+   :alt: High Level overview 
    :name: fig:debug_module_cdm
 
-   Core Debug Module
-
-The core debug module implements run-control debugging for a processor
-core. The implementation is to a certain degree core-dependent, but a
-generic implementation is sketched in @fig:debug\_module\_cdm. It has a
-memory mapped interface as described above. The debug control, status
-information and core register are mapped in memory regions. The
-run-control debugger (e.g., gdb) then sends register access requests. In
-case of a debug event (breakpoint hit) ``interrupt`` signals are
-asserted. As a reaction the CDM reads a defined address and the
-core-specific part of the CDM generates a debug event.
-
-Of course, other implementations are possible or may be required
-depending on the interface processor implementation.
-
-.. todo::
-  The specification is TBD.
-
-System Interface
-----------------
-
-The core is connected as a slave on this interface:
-
- Signal | Driver | Width | Description
- ------ | ------ | ----- | -----------
- `stall` | Module | 1 | Stall the core
- `breakpoint` | Core | 1 | Indicates breakpoint
- `strobe` | Module | 1 | Access to the core debug interface
- `ack` | Core | 1 | Complete access to the core
- `adr` | Module | ? | Address of CPU register to access
- `write` | Module | 1 | Write access
- `data_in` | Module | `DATA_WIDTH` | Write data
- `data_out` | Core | `DATA_WIDTH` | Read data
+   High Level Overview 
+
+The core debug module implements run-control debugging for OR1K CPU core. 
+The module has a memory mapped interface. CDM-OR1K maps each 32-bit wide 
+debug register of OR1K debug unit into two 16-bit wide OSD registers. 
+The run-control debugger, i.e. GDB sends register access requests to the module through OSD debug network. 
+The CDM-OR1K then transfers the corresponding data values from OSD specific 
+registers to OR1K debug unit over the debugging signals specified in :doc:`System Interface <systemif>`. 
+
+In case of a debug event (CPU debug stall event), ``interrupt`` signals are asserted. 
```

Comment:
Actually, this was the original description of the module. And I was confused about this part myself.
I think the correct wording could be: "In case of a debug event (CPU debug stall event), the CPU stalls and the control passes to GDB. The module reads/writes a defined address as per the request made by GDB".